### PR TITLE
[Fix #11405] Fix Style/WhileUntilModifier with ruby 3.1

### LINKED ISSUE #11405: undefined method `range_between' for Style/WhileUntilModifier
https://github.com/rubocop/rubocop/commit/37aa05bbac0072b5bb7df83d72949d533de58193 introduced a regression on the `Style/WhileUntilModifier` cop:

```
undefined method `range_between' for #<RuboCop::Cop::Style::WhileUntilModifier:0x00000001059f9740...
```

--------

## Expected behavior

No error

## Actual behavior

```
range_between(if_body.loc.expression.begin_pos, if_body.loc.selector.end_pos).source
        ^^^^^^^^^^^^^
Did you mean?  range_by_lines
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/mixin/statement_modifier.rb:67:in `method_source'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/mixin/statement_modifier.rb:60:in `if_body_source'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/mixin/statement_modifier.rb:51:in `to_modifier_form'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/mixin/statement_modifier.rb:46:in `length_in_modifier_form'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/mixin/statement_modifier.rb:40:in `modifier_fits_on_single_line?'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/mixin/statement_modifier.rb:16:in `single_line_as_modifier?'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/style/while_until_modifier.rb:43:in `on_while'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:101:in `public_send'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:101:in `block (2 levels) in trigger_responding_cops'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:165:in `with_cop_error_handling'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:100:in `block in trigger_responding_cops'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:99:in `each'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:99:in `trigger_responding_cops'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:69:in `on_while'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:158:in `on_block'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:71:in `on_block'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:154:in `on_def'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:71:in `on_def'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:154:in `on_class'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:71:in `on_class'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:138:in `on_while'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:71:in `on_module'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:138:in `on_while'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:71:in `on_module'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:138:in `on_while'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:71:in `on_module'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `block in on_dstr'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `each'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:137:in `on_dstr'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:71:in `on_begin'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-ast-1.24.1/lib/rubocop/ast/traversal.rb:20:in `walk'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/commissioner.rb:87:in `investigate'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/team.rb:154:in `investigate_partial'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cop/team.rb:97:in `investigate'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:322:in `inspect_file'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:266:in `block in do_inspection_loop'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:300:in `block in iterate_until_no_changes'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:293:in `loop'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:293:in `iterate_until_no_changes'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:262:in `do_inspection_loop'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:143:in `block in file_offenses'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:168:in `file_offense_cache'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:142:in `file_offenses'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:133:in `process_file'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:114:in `block in each_inspected_file'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:113:in `each'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:113:in `reduce'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:113:in `each_inspected_file'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:99:in `inspect_files'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/runner.rb:52:in `run'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cli/command/execute_runner.rb:26:in `block in execute_runner'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cli/command/execute_runner.rb:52:in `with_redirect'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cli/command.rb:11:in `run'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cli/environment.rb:18:in `run'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cli.rb:72:in `run_command'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cli.rb:79:in `execute_runners'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/lib/rubocop/cli.rb:48:in `run'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/exe/rubocop:19:in `block in <top (required)>'
/Users/user/.rvm/rubies/ruby-3.1.1/lib/ruby/3.1.0/benchmark.rb:311:in `realtime'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/gems/rubocop-1.42.0/exe/rubocop:19:in `<top (required)>'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/bin/rubocop:25:in `load'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/bin/rubocop:25:in `<main>'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/bin/ruby_executable_hooks:22:in `eval'
/Users/user/.rvm/gems/ruby-3.1.1@sorare/bin/ruby_executable_hooks:22:in `<main>'
```

## Steps to reproduce the problem

Have a while block with a long condition:

```
while players_from_first_card_scarcity.present? && current_time <= batch_settings.to
  pick_auctions(players_from_first_card_scarcity:, players:)
end
```

## RuboCop version

```
$ [bundle exec] rubocop -V
1.42.0 (using Parser 3.1.3.0, rubocop-ast 1.24.1, running on ruby 3.1.1) [arm64-darwin22]
  - rubocop-graphql 0.14.6
  - rubocop-performance 1.15.2
  - rubocop-rails 2.17.4
  - rubocop-rspec 2.13.2
  - rubocop-sorare 0.1.0
```

### PR DESCRIPTION
**Pull request by such**
Fixes https://github.com/rubocop/rubocop/issues/11405

-----------------

Before submitting the PR make sure the following are checked:

* [x ] The PR relates to *only* one subject with a clear title and description in grammatically correct, complete sentences.
* [ x] Wrote [good commit messages][1].
* [x] Commit message starts with `[Fix #issue-number]` (if the related issue exists).
* [x] Feature branch is up-to-date with `master` (if not - rebase it).
* [x] Squashed related commits together.
* [x] Added tests.
* [x] Ran `bundle exec rake default`. It executes all tests and runs RuboCop on its own code.
* [x] Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

[1]: https://chris.beams.io/posts/git-commit/

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 
 ## master (unreleased)
 
+### Bug fixes
+
+* [#11405](https://github.com/rubocop/rubocop/issues/11405): Fix undefined method `range_between' for Style/WhileUntilModifier. ([@such][])
```
Comment (koic) [inline CHANGELOG.md:7]: Can you write the changelog entry with the below instead?

> Added an entry (file) to the [changelog folder](https://github.com/rubocop/rubocop/blob/master/changelog/) named `{change_type}_{change_description}.md` if the new code introduces user-observable changes. See [changelog entry format](https://github.com/rubocop/rubocop/blob/master/CONTRIBUTING.md#changelog-entry-format) for details.

Comment (such) [reply CHANGELOG.md]: Done!

File: CHANGELOG.md [hunk 2]
```diff
@@ -6742,3 +6742,4 @@
 [@arika]: https://github.com/arika
 [@soroktree]: https://github.com/soroktree
 [@alexevanczuk]: https://github.com/alexevanczuk
+[@such]: https://github.com/such
```
Comment (koic) [inline CHANGELOG.md:6745]: Can you remove it? Because it is automatically generated by release task.

Comment (such) [reply CHANGELOG.md]: Done!

**Comment by koic** [comment]
Thanks!

**Comment by such** [comment]
No problem. When do you think this can be released?

**Comment by bbatsov** [comment]
Probably before the end of this week.

**Pull request merged**